### PR TITLE
added authSource

### PR DESCRIPTION
**Pull request by SamuelTJackson**
Hi,
I had to customize sth-comet so that it could communicate with our MongoDBs. Therefore I added the possibility to pass Auth Source with it. I tested it with our database and it works. Maybe it's interesting for you too.
New Environment variable:

DB_AUTH_SOURCE

### CODE REVIEW

File: lib/database/sthDatabase.js [hunk 1]
```diff
@@ -81,14 +81,17 @@ function getJSONCSVOptions(attrName) {
...
  */
 function connect(params, callback) {
     connectionURL =
         // prettier-ignore
-        'mongodb://' + params.authentication + '@' + params.dbURI + '/' + params.database + 
-            (params.replicaSet ? '?replicaSet=' + params.replicaSet : '');
+        'mongodb://' + params.authentication + '@' + params.dbURI + '/' + params.database +
+            (params.replicaSet ? '?replicaSet=' + params.replicaSet : '') +
+            (params.authSource ? (params.replicaSet ? '&authSource=' +
+                params.authSource : '?authSource=' + params.authSource) : '');
```
Comment (fgalan) [inline lib/database/sthDatabase.js:94]: I understand this logic ensures that only if authSource has been defined a `authSource=xxxx` token is added to the MongoDB URI.

What about `authSource == ''` (the empty string)? Would be `authSource` omitted from URI in this case? Or a `authSource=` should be added (which I understand is a wrong URI and should be avoided)?

**Comment by fgalan** [comment]
Thank you very much for the contribution!

In order to be merged, some minor aspects need to be covered (apart from the one I have provided in the code itself):

* An entry in CHANGES_NEXT_RELASE describing the change in this PR should be include. For instance "- Add: MongoDB auth source configuration (env var DB_AUTH_SOURCE)
* The new env var should be added to the list in https://github.com/telefonicaid/fiware-sth-comet/blob/master/doc/manuals/running.md#running-the-sth-server, along with the other `DB_` env vars.
* The config.js counterpart for DB_AUTH_SOURCE should be added to that file

File: lib/database/sthDatabase.js (hunk 1 above)
Comment (SamuelTJackson) [reply lib/database/sthDatabase.js:94]: As I understand the query (`params.authSource ?`) and Javascript, the string is checked for null, undefined or empty. (https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in)

Comment (fgalan) [reply lib/database/sthDatabase.js:94]: Ok. NTC

**Pull request merged**